--- rasdaman-8.4.0/configure.ac.orig	2013-03-20 13:40:38.000000000 +0100
+++ rasdaman-8.4.0/configure.ac	2014-05-11 16:11:58.382902633 +0200
@@ -229,7 +229,7 @@
 AC_ARG_WITH([pic],
             [AS_HELP_STRING([--with-pic], 
                             [generate position independent code (PIC)])],
-            [CXXFLAGS+=" -fPIC " CFLAGS+=" -fPIC "],
+            [CXXFLAGS="$CXXFLAGS -fPIC " CFLAGS="$CFLAGS -fPIC "],
             [])
 
 # postgres is automatically defined in m4/ct_check_postgres_db.m4
@@ -247,7 +247,7 @@
 AC_ARG_ENABLE([benchmark],
               [AS_HELP_STRING([--enable-benchmark],
                               [this feature enables generation of binaries that contain extra code for benchmark output.])],
-              [CXXFLAGS+=" -DRMANBENCHMARK "],
+              [CXXFLAGS="$CXXFLAGS -DRMANBENCHMARK "],
               [])
 
 AC_ARG_WITH([test-server],
@@ -356,7 +356,7 @@
 for DIR in ${prefix}/include /usr/local/include/ /usr/include /opt/local/include $CPATH_NO_COLON; do
     test -d "$DIR" && NETPBM_PPMH=`find $DIR -name ppm.h`
     if test -n "$NETPBM_PPMH"; then
-      CXXFLAGS+=" -I ${NETPBM_PPMH%%ppm.h} "
+      CXXFLAGS="$CXXFLAGS -I ${NETPBM_PPMH%%ppm.h} "
       AC_MSG_RESULT([yes])
       break
     fi
@@ -369,13 +369,11 @@
 AS_IF([test "x$RMANDEBUG" != x],
       [AC_DEFINE([RMANDEBUG])
        AC_DEFINE([DEBUG])
-       CXXFLAGS+=" -g3 -gdwarf-2 -O0 "
+       CXXFLAGS="$CXXFLAGS -g3 -gdwarf-2 -O0 "
       ], 
       [
        AC_DEFINE([__NO_MATH_INLINES])
-       CXXFLAGS+=" -O2 "
       ])
-# CXXFLAGS+= -fPIC  ???NEEDED???
 AC_DEFINE_UNQUOTED([BASEDBSTRING], ["$BASEDBSTRING"])
 AC_DEFINE([EARLY_TEMPLATE])    
 AC_C_BIGENDIAN([AC_DEFINE([IS_BIG_ENDIAN])], [AC_DEFINE([IS_LITTLE_ENDIAN])])
@@ -384,18 +382,14 @@
 AC_DEFINE([RASARCHITECTURE], ["X86"])
 AC_DEFINE([RASSCHEMAVERSION], [5])
 AC_DEFINE([NOPRE], [], [Description])
-# if not on DEC Alpha
-BASEDBLDFLAGS+=" -lnsl"
-STATICBASEDBLDFLAGS+=" -lnsl"
-# endif
 AC_DEFINE([NO_in_addr_t])
-BASEDBCXXFLAGS+=" $DBIF_INC "
+BASEDBCXXFLAGS="$BASEDBCXXFLAGS $DBIF_INC "
 
 COMPDATE=`date +"%d.%m.%Y %H:%M:%S"`
 AC_DEFINE_UNQUOTED([COMPDATE], ["$COMPDATE"])
 
-CXXFLAGS+=" -I$abs_top_srcdir -I$abs_top_srcdir/debug -I$abs_top_srcdir/include $USER_CXXFLAGS"
-CFLAGS+=" $USER_CFLAGS"
+CXXFLAGS="$CXXFLAGS -I$abs_top_srcdir -I$abs_top_srcdir/debug -I$abs_top_srcdir/include $USER_CXXFLAGS"
+CFLAGS="$CFLAGS $USER_CFLAGS"
 
 CLIENTCXXFLAGS=$CXXFLAGS
 CLIENTLDFLAGS=$CXXLDFLAGS
